<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>alpha diversity, properly â€¢ breakaway</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="alpha diversity, properly">
<meta property="og:description" content="">
<meta property="og:image" content="http://adw96.github.io/breakaway/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">breakaway</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">4.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/breakaway.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/alpha-better.html">alpha diversity, properly</a>
    </li>
    <li>
      <a href="../articles/betta-figure.html">Comparing samples visually with betta</a>
    </li>
    <li>
      <a href="../articles/extended-tutorial.html">Getting into the weeds</a>
    </li>
    <li>
      <a href="../articles/phyloseq.html">breakaway + phyloseq</a>
    </li>
    <li>
      <a href="../articles/resample-estimates.html">Variance of alpha diversity estimates</a>
    </li>
    <li>
      <a href="../articles/sample-size-calculations.html">Sample size calculations for alpha diversity</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>alpha diversity, properly</h1>
                        <h4 class="author">Amy Willis</h4>
            
            <h4 class="date">2018-05-16</h4>
      
      
      <div class="hidden name"><code>alpha-better.Rmd</code></div>

    </div>

    
    
<div id="introduction-to-estimating-alpha-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-to-estimating-alpha-diversity" class="anchor"></a>Introduction to estimating alpha diversity</h2>
<p>It is most often the case that only a small fraction of the entire microbial population of interest is sampled in microbiome studies. While scientists are typically interested in the entire microbial population, they only have some small sample which which represents it. Unfortunately it is typical to not adjust for the inexhaustive experiment, and to analyse only the sequences that were observed without any adjustment for missing data. This practice is no more pronounced than in the setting of alpha diversity.</p>
<p>Alpha diversity is special because common alpha diversity estimates (such as plug-in and rarefied estimates) are highly dependent on sample size. For example, increasing the number of sequences (library size) generally increases the diversity. This is paradoxical, because true diversity of the ecosystem exists regardless of the experiment, and sampled ecosystems. The mission of <code>breakaway</code> is to provide the statistical tools to adjust for inexhaustive sampling in microbiome experiments.</p>
<p>This tutorial covers using <code>breakaway</code> to estimate the Shannon index when sequencing or sampling was inexhaustive and taxa may be missing.</p>
</div>
<div id="standard-errors-on-shannon-index-estimates" class="section level2">
<h2 class="hasAnchor">
<a href="#standard-errors-on-shannon-index-estimates" class="anchor"></a>Standard errors on Shannon index estimates</h2>
<p>Load <code>breakaway</code> and have a look at one of the in-built datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(breakaway)
<span class="kw">head</span>(apples)</code></pre></div>
<pre><code>##   index frequency
## 1     1       277
## 2     2       159
## 3     3       128
## 4     4        59
## 5     5        42
## 6     6        51</code></pre>
<p>Estimate the Shannon index of the population that the <code>apples</code> sample was drawn from!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/shannon_better.html">shannon_better</a></span>(apples)</code></pre></div>
<pre><code>##     index estimate standard_error   lower    upper
## 1 shannon 6.551956     0.05263679 6.44879 6.655122</code></pre>
<p>Cool! What about when we have a full dataset? How should this be plotted?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#apply(toy_otu_table, 2, shannon_better)</span></code></pre></div>
<p>TODO: Finish this!</p>
<!-- Let's start by writing some functions to invent fake communities. This is so we can validate the idea on fake data. -->
<!-- ```{r} -->
<!-- library(breakaway) -->
<!-- library(ggplot2) -->
<!-- library(gridExtra) -->
<!-- library(dplyr) -->
<!-- library(magrittr) -->
<!-- library(grid) -->
<!-- library(lattice) -->
<!-- # create a total community -->
<!-- make_community2 <- function(CC, alpha_parameter = 1, beta_parameter = 2) { -->
<!--   relative_proportions <- rbeta(CC, alpha_parameter, beta_parameter) # this is just a good way to get relative abundances -->
<!--   raw_abundances <- relative_proportions/sum(relative_proportions) -->
<!--   output <- list() -->
<!--   output$proportions <- raw_abundances -->
<!--   output -->
<!-- } -->
<!-- # let's simulate some rarefaction curves -->
<!-- subsample_community <- function(population, n) { -->
<!--   output <- list() -->
<!--   who <- sample(population$names, n, replace = F) -->
<!--   community <- c(as.matrix(table(who))) -->
<!--   abundances <- community/sum(community) -->
<!--   output$names <- who -->
<!--   output$proportions <- abundances -->
<!--   output -->
<!-- } -->
<!-- draw_rarefaction <- function(population, pts = NA, replicates = 5, ecosystem = NA) { -->
<!--   rarefaction_table <- data.frame() -->
<!--   if (all(is.na(pts))) pts <- c(round(seq(from = 5, to = length(population$names)/100, length = 100)), -->
<!--                                 round(seq(from = length(population$names)/100, to = length(population$names)/10, length = 100)), -->
<!--                                 round(seq(from = length(population$names)/10, to = length(population$names)/4, length = 100)), -->
<!--                                 round(seq(from = length(population$names)/4, to = length(population$names), length = 10))) -->
<!--   who <- replicate(replicates, sample(population$names, length(population$names), replace = F)) -->
<!--   for (i in pts) { -->
<!--     subsample_diversity <- rep(NA, replicates) -->
<!--     subsample_shannon <- rep(NA, replicates) -->
<!--     for (j in 1:replicates) {   -->
<!--       subsample <- who[1:i, j] -->
<!--       subsample_diversity[j] <- length(unique(subsample)) -->
<!--       community <- c(as.matrix(table(subsample))) -->
<!--       proportions <- community/sum(community) -->
<!--       subsample_shannon[j] <- breakaway::shannon(proportions) -->
<!--     } -->
<!--     rarefaction_table %<>% rbind(data.frame("reads" = i,  -->
<!--                                             "richness" = min(subsample_diversity), "richness_max" = max(subsample_diversity),  -->
<!--                                             "shannon" = min(subsample_shannon), "shannon_max" = max(subsample_shannon),  -->
<!--                                             "ecosystem" = ecosystem, "type" = "rcurve")) -->
<!--   } -->
<!--   rarefaction_table -->
<!-- } -->
<!-- get_point <- function(ecosystem, read, label) { -->
<!--   read <- round(read) -->
<!--   my_population <- get(paste("population", ecosystem, sep="")) -->
<!--   ecosystem_name <- paste("ecosystem", ecosystem, sep="") -->
<!--   subsample <- sample(x = c(1:length(my_population$proportions)), size = read, prob = my_population$proportions, replace = T) -->
<!--   point <- data.frame("reads" = read,  -->
<!--                       "richness" = length(unique(subsample)), "richness_max" = NA, -->
<!--                       "shannon" = breakaway::shannon(c(as.matrix(table(subsample)))/sum(c(as.matrix(table(subsample))))),  -->
<!--                       "shannon_max" = NA,  -->
<!--                       "ecosystem" = ecosystem_name, "type" = "observation") -->
<!--   subsample1 <- list() -->
<!--   subsample1$names <- subsample -->
<!--   subsample1$proportions <- c(as.matrix(table(subsample)))/sum(c(as.matrix(table(subsample)))) -->
<!--   rarefied_sample <- draw_rarefaction(subsample1, replicates = 1, pts = seq(from = 1, to = read, length = 100),  -->
<!--                                       ecosystem = ecosystem_name) -->
<!--   rarefied_sample$type <- paste("rarefied_sample", label, sep="") -->
<!--   richness_estimate <- subsample %>% make_frequency_count_table %>% breakaway( output = F, answers = T, plot = F) %$% ci  -->
<!--   shannon_estimate <- subsample %>% make_frequency_count_table  %>% shannon_better -->
<!--   estimate <- data.frame("reads" = read,  -->
<!--                          "richness" = richness_estimate[1], "richness_max" = richness_estimate[2], -->
<!--                          "shannon" = shannon_estimate[1,4], "shannon_max" = shannon_estimate[1,5], -->
<!--                          "ecosystem" = ecosystem_name, "type" = paste("estimate", label, sep="")) -->
<!--   rbind(point, rarefied_sample, estimate) -->
<!-- } -->
<!-- get_truth <- function(ecosystem_number) { -->
<!--   ecosystem <- paste("population", ecosystem_number, sep = "") -->
<!--   population <- get(ecosystem) -->
<!--   data.frame("reads" = 0, "richness" = length(population$proportions), "richness_max" = NA, -->
<!--              "shannon" = breakaway::shannon(population$proportions), "shannon_max" = NA,  -->
<!--              "ecosystem" =  paste("ecosystem", ecosystem_number, sep = ""), "type" = "truth") -->
<!-- } -->
<!-- ``` -->
<!-- Now let's actually create some communities, and draw some pretend samples from them. -->
<!-- ```{r} -->
<!-- set.seed(170918) -->
<!-- Ca = 6000 -->
<!-- Cb = 5000 -->
<!-- aa = 0.5 -->
<!-- ab = aa -->
<!-- ba = 3 -->
<!-- bb = 1 -->
<!-- # rarefaction curves -->
<!-- population1 <- make_community2(Ca, alpha_parameter = aa, beta_parameter = ab) -->
<!-- population2 <- make_community2(Cb, alpha_parameter = ba, beta_parameter = bb) -->
<!-- # observations and truth -->
<!-- truth <- rbind(get_truth(1), get_truth(2)) -->
<!-- observations <- rbind(get_point(2, 25000, 1),  -->
<!--                       get_point(2, 12000, 2), -->
<!--                       get_point(2, 8000, 3),  -->
<!--                       get_point(2, 7000, 4), -->
<!--                       get_point(2, 16000, 5), -->
<!--                       get_point(1, 8000, 1), -->
<!--                       get_point(1, 14000, 2), -->
<!--                       get_point(1, 5000, 3), -->
<!--                       get_point(1, 7500, 4), -->
<!--                       get_point(1, 4500, 5), -->
<!--                       get_point(1, 3000, 6)) -->
<!-- rarefaction_level <- observations %>% filter(type == "observation") %>% select(reads) %>% min -->
<!-- rarefaction_table <- rbind(observations, truth) -->
<!-- rarefaction_table %>% select(ecosystem) %>% unlist %>% substring(10, 11) %>% as.numeric -> rarefaction_table$ecosystem_number -->
<!-- plot_base <- ggplot(data = rarefaction_table, aes(col = ecosystem)) + -->
<!--   theme_bw()  -->
<!-- plot_5_a <- plot_base +  -->
<!--   labs(x="No. reads", y="Taxonomic richness",col = "") + -->
<!--   coord_cartesian(xlim = c(0, 26000), ylim=c(0, 7000)) +  -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=reads, y=richness, col = ecosystem))+ -->
<!--   theme(legend.position = c(0.7, 0.25), legend.text=element_text(size=7))  + -->
<!--   scale_color_discrete(name="", -->
<!--                        breaks=c("ecosystem1", "ecosystem2"),  -->
<!--                        labels=c("Environment A", "Environment B")) + -->
<!--   guides(fill=FALSE) -->
<!-- plot_5_b <- plot_base +  -->
<!--   labs(x="No. reads", y="Shannon diversity",col = "") + -->
<!--   coord_cartesian(xlim=c(0, 26000), ylim=c(7, 9)) +  -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=reads, y=shannon, col = ecosystem))+ -->
<!--   theme(legend.position = c(0.7, 0.3), legend.text=element_text(size=7))  + -->
<!--   scale_color_discrete(name="", -->
<!--                        breaks=c("ecosystem1", "ecosystem2"),  -->
<!--                        labels=c("Environment A", "Environment B")) + -->
<!--   guides(fill=FALSE) -->
<!-- plot_6_a <- plot_5_a +  -->
<!--   geom_ribbon(data = rarefaction_table %>% filter(type == "rcurve"), -->
<!--               aes(x=reads, ymin=richness, ymax=richness_max, col = ecosystem, fill = ecosystem), alpha = 0, linetype = 0) + -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"), -->
<!--            aes(x=reads, y=richness, col = ecosystem))+ -->
<!--   geom_line(data = rarefaction_table %>% filter(type == "rarefied_sample1"), -->
<!--             aes(x=reads, y=richness, col = ecosystem)) + -->
<!--   ggtitle("Rarefaction curve") +  -->
<!--   geom_line(data = rarefaction_table %>% filter(type == "rarefied_sample2"), -->
<!--             aes(x=reads, y=richness, col = ecosystem))  -->
<!-- plot_6_d <- plot_5_b +  -->
<!--   geom_ribbon(data = rarefaction_table %>% filter(type == "rcurve"), -->
<!--               aes(x=reads, ymin=shannon, ymax=shannon_max, col = ecosystem, fill = ecosystem), alpha = 0, linetype = 0) + -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"), -->
<!--              aes(x=reads, y=shannon, col = ecosystem))+ -->
<!--   geom_line(data = rarefaction_table %>% filter(type == "rarefied_sample1"), -->
<!--             aes(x=reads, y=shannon, col = ecosystem)) + -->
<!--   geom_line(data = rarefaction_table %>% filter(type == "rarefied_sample2"), -->
<!--             aes(x=reads, y=shannon, col = ecosystem)) + -->
<!--   ggtitle("Rarefaction curve") -->
<!-- plot_7a_left <- plot_6_a + geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--              aes(yintercept=richness, col = ecosystem), lty = 2)  -->
<!-- plot_7b_left <- plot_6_d + geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--              aes(yintercept=shannon, col = ecosystem), lty = 2)  -->
<!-- plot_7_a <- plot_base +  -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                         seq(-0.1, 0.1, length.out = 5)), y=richness, col = ecosystem), alpha = 0.1) +  -->
<!--   geom_linerange(data = rarefaction_table %>% filter(substr(type, 1, 8) == "estimate"),  -->
<!--                  aes(x = ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                               seq(-0.1, 0.1, length.out = 5)), ymin=richness, ymax = richness_max), lty = 1) +  -->
<!--   geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--              aes(yintercept=richness, col = ecosystem), lty = 2) + -->
<!--   labs(y="Taxonomic richness",col = "") + -->
<!--   theme(legend.position="none") + -->
<!--   ggtitle("Estimated true richness with error") +  -->
<!--   coord_cartesian(ylim=c(0, 7000)) -->
<!-- plot_7_b <- plot_base +  -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                         seq(-0.1, 0.1, length.out = 5)), y=shannon, col = ecosystem), alpha = 0.1) +  -->
<!--   geom_linerange(data = rarefaction_table %>% filter(substr(type, 1, 8) == "estimate"),  -->
<!--                  aes(x = ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                               seq(-0.1, 0.1, length.out = 5)), ymin=shannon, ymax = shannon_max), lty = 1) +  -->
<!--   geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--              aes(yintercept=shannon, col = ecosystem), lty = 2) + -->
<!--   labs(y="Shannon index",col = "") + -->
<!--   theme(legend.position="none") + -->
<!--   ggtitle("Estimated true Shannon with error") +  -->
<!--   coord_cartesian(ylim=c(7, 9)) -->
<!-- plot_14_a <- plot_base +  -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   geom_linerange(data = rarefaction_table %>% filter(substr(type, 1, 8) == "estimate"),  -->
<!--                  aes(x = ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                               seq(-0.1, 0.1, length.out = 5)), ymin=richness, ymax = richness_max), lty = 1) +  -->
<!--   labs(y="Taxonomic richness",col = "") + -->
<!--   theme(legend.position="none") + -->
<!--   ggtitle("Estimated true richness with error") +  -->
<!--   coord_cartesian(ylim=c(0, 7000)) -->
<!-- plot_14_b <- plot_base +  -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   geom_linerange(data = rarefaction_table %>% filter(substr(type, 1, 8) == "estimate"),  -->
<!--                  aes(x = ecosystem_number + c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                               seq(-0.1, 0.1, length.out = 5)), ymin=shannon, ymax = shannon_max), lty = 1) +  -->
<!--   labs(y="Shannon index",col = "") + -->
<!--   theme(legend.position="none") + -->
<!--   ggtitle("Estimated true Shannon with error") +  -->
<!--   coord_cartesian(ylim=c(7, 9)) -->
<!-- plot_6_b <- plot_base + -->
<!--   labs(y="Taxonomic richness",col = "") + -->
<!--   coord_cartesian(ylim=c(0, 7000)) +  -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=ecosystem_number+ c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                        seq(-0.1, 0.1, length.out = 5)), y=richness, col = ecosystem)) +  -->
<!--   # geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--   #            aes(yintercept=richness, col = ecosystem), lty = 2) + -->
<!--   ggtitle("Raw richness") + -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   theme(legend.position="none")  -->
<!-- plot_6_e <- plot_base +   -->
<!--   labs(y="Shannon diversity",col = "") + -->
<!--   scale_x_continuous(name="Environment", breaks=c(1,2), labels=c("A", "B"), limits = c(0.5, 2.5)) + -->
<!--   geom_point(data = rarefaction_table %>% filter(type == "observation"),  -->
<!--              aes(x=ecosystem_number+ c(seq(-0.1, 0.1, length.out = 6),  -->
<!--                                        seq(-0.1, 0.1, length.out = 5)), y=shannon, col = ecosystem)) +  -->
<!--   # geom_hline(data = rarefaction_table %>% filter(type == "truth"),  -->
<!--   #            aes(yintercept=shannon, col = ecosystem), lty = 2) + -->
<!--   theme(legend.position="none") + -->
<!--   ggtitle("Raw Shannon index") + -->
<!--   coord_cartesian(ylim=c(7, 9))  -->
<!-- grid.arrange(grobs = list(plot_6_b, plot_6_e, plot_14_a, plot_14_b),  -->
<!--              layout_matrix =  matrix(c(1, 2, 3, 4), nrow = 2)) -->
<!-- ``` -->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction-to-estimating-alpha-diversity">Introduction to estimating alpha diversity</a></li>
      <li><a href="#standard-errors-on-shannon-index-estimates">Standard errors on Shannon index estimates</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Amy Willis, Kathryn Barger, John Bunge.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
